pkg/p2p/libp2p/libp2p.go

```diff
@@ -193,13 +193,21 @@ func New(ctx context.Context, o Options) (*Service, error) {
 		}
 
 		s.peers.add(stream.Conn(), i.Address)
-		s.addrssbook.Put(i.Address, stream.Conn().RemoteMultiaddr())
+		remoteMultiaddr, err := ma.NewMultiaddr(fmt.Sprintf("%s/p2p/%s", stream.Conn().RemoteMultiaddr().String(), peerID.Pretty()))
+		if err != nil {
+			s.logger.Debugf("multiaddr error: handle %s: %v", peerID, err)
```

Comment:
this is quite different message than what is in error

Reply:
The idea is that message in error is user scope, and debug is a developer scope which has a more detailed description. In this case, for the user, the result is that connection failed. For the developer is that it failed because because of the multiaddr generation error. 
This is the approach me & @janos took across the bee, you can see it across the project, and I believe that we all talked about it, but I might be wrong. If you don't like it, let's discuss it and switch to something else maybe?